.github/workflows/integration.yml

```diff
@@ -0,0 +1,45 @@
+name: Integration
+
+on:
+  push:
+    branches: 
+      - main
+    tags:
+      - '*'
+  pull_request:
+    branches: 
+      - main
+
+jobs:
+  build:
+    runs-on: minafoundation-default-runners
+  
+    steps:
+    - name: 📥 Checkout
+      uses: actions/checkout@v3    
+
+    - name: 📥 Update Git Submodules
+      run: |
+        git submodule init
+        git submodule update
+
+    - name: 📦 Install Nix
+      uses: cachix/install-nix-action@v22
+      with:
+        nix_path: nixpkgs=channel:nixos-unstable
+
+    - name: 🤌 Get Minimina
+      uses: MinaFoundation/install-minimina-action@v1
+      with:
+        stream: stable
+
+    - name: 🛠️ Build Docker
+      env:
+        IMAGE_NAME: uptime-service-backend
+        TAG: integration-test
+      run: nix-shell --run "make docker"
+
+    - name: 🧪 Run Integration Tests
+      env:
+        UPTIME_SERVICE_SECRET: ${{ secrets.UPTIME_SERVICE_SECRET }}
```

Comment:
What is this secret used for? The naming is very generic, could be the private key, the password or any other secret for that matter

Reply:
The secret is a passphrase used for decoding uptime service config files (e.g. https://github.com/MinaFoundation/uptime-service-backend/blob/integration-test/src/integration_tests/constants.go#L61-L64) The encoded configuration is kept here (https://github.com/MinaFoundation/uptime-service-backend/tree/integration-test/test/integration/topology/uptime_service_config). There is also convenient `encodeSecrets()` that can be used to re-encode configs in case they need to be changed.

`UPTIME_SERVICE_SECRET` is kept as a secret in the https://github.com/MinaFoundation/uptime-service-backend/settings/secrets/actions.

Reply:
I've added a readme note on how to run integration tests and also renamed encoding/decoding methods. Hopefully it will make things more clear. Let me know. :+1:

Reply:
I think a oneliner comment here would be good, just to know what this is and what the purpose of it is